src/shared/ask-password-api.c

```diff
@@ -153,7 +174,7 @@ static int add_to_keyring(const char *keyname, AskPasswordFlags flags, char **pa
 
         if (keyctl(KEYCTL_SET_TIMEOUT,
                    (unsigned long) serial,
-                   (unsigned long) DIV_ROUND_UP(KEYRING_TIMEOUT_USEC, USEC_PER_SEC), 0, 0) < 0)
+                   (unsigned long) DIV_ROUND_UP(keyring_timeout(), USEC_PER_SEC), 0, 0) < 0)
```

Comment:
hmm, keyring_timeout() should probably be treated like ASK_PASSWORD_PUSH_CACHE being off, and caught at the top of the function